CardGetters.php

```diff
@@ -318,7 +318,8 @@ function &GetAllies($player)
 function &GetTheirAllies($player)
 {
   global $p1Allies, $p2Allies;
-  return $player == 1 ? $p2Allies : $p1Allies;
```

Comment:
what was wrong with this ternary operator?

Reply:
The PHP code was issuing a warning: "Only variable references should be returned by reference." For some reason, when using the ternary operator, it didn't recognize that the return was a variable reference.

Reply:
warning in your editor or in the browser?

Reply:
Warning in the debugger

Reply:
hm.. that's odd. i'm not getting any warning in VSCode

Reply:
Are you running the debug mode on VSCode?

Reply:
yes

Reply:
You can replicate the warning here https://onlinephp.io/

Example:
```
<?php

function &GetTheirAllies($player) {
  $p1Allies = [1, 2, 3];
  $p2Allies = [4, 5, 6];
  return $player == 1 ? $p2Allies : $p1Allies;
}

$var1 = &GetTheirAllies(1);

echo($var1[0]);
```

![image](https://github.com/user-attachments/assets/18f697cb-bc38-44b1-b7e2-5d0365a75b9d)

Reply:
ahh okay. interesting..